paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.cc

```diff
@@ -0,0 +1,219 @@
+             input_names.end() &&
+         conv_op->Op()->Input("Bias").size() > 0;
+}
+
+bool ShouldSkipConv(ir::Node* conv_op, Scope* scope, ir::Node* conv_filter) {
+  if (!platform::HasOpINT8DataType(conv_op->Op())) {
+    return true;
+  }
+
+  auto filter_var = scope->GetVar(conv_filter->Name());
+  if (filter_var->Get<LoDTensor>().dtype() != phi::DataType::FLOAT32) {
+    VLOG(4) << "Skipping convolution (id: " << conv_op->id()
+            << ") because it's a bug that it is detected again.";
+    return true;
+  }
+
+  VLOG(4) << "Not skipping convolution (id: " << conv_op->id() << ")";
+  return false;
+}
+
+void QuantizeConvFilter(Scope* scope, ir::Graph* g, ir::Node* conv_op,
+                        ir::Node* conv_filter) {
+  const auto scale_weights =
+      conv_op->Op()->GetAttrIfExists<std::vector<float>>("Scale_weights");
+
+  auto* weight_tensor =
+      scope->GetVar(conv_filter->Name())->GetMutable<LoDTensor>();
+
+  QuantizeParams<int8_t>(weight_tensor, scale_weights);
+
+  conv_op->Op()->SetAttr("Scale_weights", std::vector<float>(1, 1));
+}
+
+void QuantizeConvBias(Scope* scope, ir::Graph* g, ir::Node* conv_op) {
+  const auto bias_scales =
+      conv_op->Op()->GetAttrIfExists<std::vector<float>>("Bias_scales");
+
+  std::string conv_bias_name = conv_op->Op()->Input("Bias")[0];
+  auto* bias_tensor = scope->GetVar(conv_bias_name)->GetMutable<LoDTensor>();
+
+  QuantizeParams<int32_t>(bias_tensor, bias_scales);
+
+  conv_op->Op()->SetAttr("Bias_scales", std::vector<float>(1, 1));
+}
+
+}  // namespace
+
+ParamsQuantizationMkldnnPass::ParamsQuantizationMkldnnPass() {
+  AddOpCompat(OpCompat("conv2d"))
+      .AddInput("Input")
+      .IsTensor()
+      .End()
+      .AddInput("Filter")
+      .IsTensor()
+      .End()
+      .AddInput("Bias")
+      .IsTensor()
+      .IsOptional()
+      .End()
+      .AddInput("ResidualData")
+      .IsTensor()
+      .IsOptional()
+      .End()
+      .AddOutput("Output")
+      .IsTensor()
+      .End()
+      .AddAttr("strides")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("paddings")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("padding_algorithm")
+      .IsOptional()
+      .IsStringIn({"EXPLICIT", "SAME", "VALID"})
+      .End()
+      .AddAttr("groups")
+      .IsNumGE(1)
+      .End()
+      .AddAttr("dilations")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("data_format")
+      .IsStringIn({"NCHW", "AnyLayout"})
+      .End();
+}
+
+void ParamsQuantizationMkldnnPass::Conv(ir::Graph* graph,
+                                        bool with_residual_data) const {
+  GraphPatternDetector gpd;
+  patterns::ConvResidual conv_pattern(gpd.mutable_pattern(), name_scope_);
+  conv_pattern(with_residual_data);
+
+  int params_to_int8_conv_found = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    if (!IsCompat(subgraph, g)) {
```

Comment:
this function is risky since we are modifying ops out of spec.

Reply:
What do you mean?

Reply:
AFAIK This function compare an operator with the prototxt which define what is valid.
It can fail if someone add any custom attribute etc.

Reply:
@lidanqing-intel can I get rid of this IsCompat?

Reply:
I see not all of our mkldnn passes have this check, but maybe better make sure we can skip it.